### DISCUSSION TITLE
Is there any way to read the texture in the vertex shader?

### DISCUSSION THREAD

**Discussion by redcamel** [gpuweb/gpuweb#4560] [Q&A]
Is there any way to read the texture in the vertex shader?

**Comment by redcamel** [comment]
had to do was use textureSampleLevel. -_-

**Comment by greggman** [comment]
yes, use any texture function that doesn't rely on derivatives since derivatives, the rate of change of inter-stage variables, only exist in fragment shaders.. `textureSampleLevel`, `textureSampleGrad`, `textureLoad`, `textureGather` all work in vertex shaders.
